biodatasets/bio_sim_verb/bio_sim_verb.py

```diff
@@ -0,0 +1,175 @@
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="bio_sim_verb_source",
+            version=SOURCE_VERSION,
+            description="bio_sim_verb source schema",
+            schema="source",
+            subset_id="bio_sim_verb",
+        ),
+        BigBioConfig(
+            name="bio_sim_verb_bigbio_pairs",
+            version=BIGBIO_VERSION,
+            description="bio_sim_verb BigBio schema",
+            schema="bigbio_pairs",
+            subset_id="bio_sim_verb",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "bio_sim_verb_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "text_1": datasets.Value("string"),
+                    "text_2": datasets.Value("string"),
+                    "label": datasets.Value("string"),
+                }
+            )
+
+        # Using in pairs schema
+        elif self.config.schema == "bigbio_pairs":
+            features = schemas.pairs_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION, features=features, homepage=_HOMEPAGE, license=_LICENSE, citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # Whatever you put in gen_kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": data_dir, "split": "train"},
+            )
+        ]
+
+    def _generate_examples(self, filepath, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        with open(filepath, encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(
+                csv_file, quotechar='"', delimiter="\t", quoting=csv.QUOTE_ALL, skipinitialspace=True
+            )
+
+            if self.config.schema == "source":
+                for id_, row in enumerate(csv_reader):
+                    text_1, text_2, label = row
+                    yield id_, {
+                        "text_1": text_1,
+                        "text_2": text_2,
+                        "label": label,
+                    }
+
+            elif self.config.schema == "bigbio_pairs":
+                uid = 0
+                for id_, row in enumerate(csv_reader):
+                    uid += 1
+                    text_1, text_2, label = row
+                    yield id_, {
+                        "id": str(uid),
+                        "document_id": "NULL",
+                        "text_1": text_1,
+                        "text_2": text_2,
+                        "label": "0" if float(label) < 5 else "1",
```

Comment:
haha, here I was actually thinking just the string representation of the float (will have to go back through my PR comments but I think I suggested that in a different `bio_sim_X` PR)

Reply:
ok, yep, it was @tanmaylaud 's PR for `bio-simlex` #406 :) yea .... I think I'm still in the "use float for source schema" camp ... but @debajyotidatta what do you think, is this thresholding how things were handled in the prompt source work?

Reply:
@debajyotidatta I'd prefer we don't threshold the dataset. The users of the dataloader will expect the dataset to preserve as much streamlined information as possible. There is no guarantee this transformation actually preserves a well-balanced binary label class, and subtleties like scores close to 5 (i.e if the score distribution is gaussian in the center of the score range) will be lost.

Let's please do float for the source schema and str for the bigbio schema (to adhere to our schema unit-checks).